cli/client.go

```diff
@@ -380,6 +387,111 @@ func RobotsPartShellAction(c *cli.Context) error {
 	)
 }
 
+// checkUpdateResponse holds the values used to hold release information.
+type checkUpdateResponse struct {
+	Name       string `json:"name"`
+	TagName    string `json:"tag_name"`
+	TarballURL string `json:"tarball_url"`
+}
+
+// CheckUpdateAction is the corresponding Action for 'check-update'.
+func CheckUpdateAction(c *cli.Context) error {
+	if c.Bool(quietFlag) {
+		return nil
+	}
+
+	dateCompiledRaw := rconfig.DateCompiled
+	dateCompiled, err := time.Parse("2006-01-02", dateCompiledRaw)
+	if err != nil {
+		// user installed from go install or source
+		utils.UncheckedError(err)
+		return nil
+	}
+
+	conf, err := configFromCache()
+	if err != nil {
+		if !os.IsNotExist(err) {
+			utils.UncheckedError(err)
+			return nil
+		}
+		conf = &config{}
+	}
+
+	if conf.LastUpdateCheck == "" {
+		conf.LastUpdateCheck = time.Now().Format("2006-01-02")
+	  err := storeConfigToCache(conf)
+    if err != nil { 
+			utils.UncheckedError(err)
+    }
+	} else {
+		lastCheck, err := time.Parse("2006-01-02", conf.LastUpdateCheck)
+		if err != nil {
+			// corrupted date
+			utils.UncheckedError(err)
+			return nil
+		}
+		if time.Since(lastCheck) < time.Hour*24*7 {
+			return nil
+		}
+	}
+
+	// compare to current version
+	appVersion := rconfig.Version
+
+	// skip check if on stable and install is older than two week
+	if appVersion != "" && time.Since(dateCompiled) < time.Hour*24*7*2 {
+		return nil
+	}
+
+	// skip check if on dev/latest and install is less than six weeks old
+	if appVersion == "" && time.Since(dateCompiled) < time.Hour*24*7*6 {
```

Comment:
This is similar to Max's question earlier. 

At the moment, if built from source or go-install, there is no version info stored in the binary. If installing via brew or curl, it will have this info (ie v0.1.0, etc). 

If you use the `version` command, the logic will check for this version info. If the info is an empty string, it outputs `(dev)` as a placeholder.

Reply:
would put all that info in a comment, or factor out the logic in `VersionAction` and check for `(dev)` instead

Reply:
I've made some changes to the Makefile to automatically set the version info to (dev) instead of leaving an empty string.
This will remove a couple steps and make things more clear.